### PR TITLE
Add UpdateOpenGraphMetaTags to adapter

### PR DESCRIPTION
**Pull request by kapillamba4**
Part of #809 

### Description
UpdateOGMetaTags sqldb adapter method returns new ShortLink after updating OpenGraph MetaTags.
A corresponding GraphQL mutation API will be created for updating OpenGraph MetaTags for a ShortLink

### CODE REVIEW

File: backend/app/adapter/sqldb/short_link_integration_test.go [hunk 1]
```diff
@@ -47,6 +47,102 @@ type shortLinkTableRow struct {
...
+
+	testCases := []struct {
+		name              string
+		tableRows         []shortLinkTableRow
+		alias             string
+		metaTags          metatag.OpenGraph
+		expectedShortLink entity.ShortLink
+	}{
+		{
+			name: "Update nil meta tags",
```
Comment (byliuyang) [inline backend/app/adapter/sqldb/short_link_integration_test.go:66]:
```suggestion
			name: "Twitter tags not provided",
```

File: backend/app/adapter/sqldb/short_link_integration_test.go [hunk 2]
```diff
@@ -47,6 +47,102 @@ type shortLinkTableRow struct {
...
+				LongLink: "http://www.google.com",
+				OpenGraphTags: metatag.OpenGraph{
+					Title:       &title1,
+					Description: &description1,
+					ImageURL:    &imageURL1,
+				},
+			},
+		},
+		{
+			name: "Update non nil meta tags",
```
Comment (byliuyang) [inline backend/app/adapter/sqldb/short_link_integration_test.go:90]:
```suggestion
			name: "Twitter tags provided",
```

Comment (byliuyang) [reply backend/app/adapter/sqldb/short_link_integration_test.go:90]: Please add test case for Open Graph Tags also

**Comment by byliuyang** [comment]
Hi @kapillamba4 , thank you so much for working on this feature! I left few comments in the PR.

File: backend/app/adapter/sqldb/short_link.go [hunk 3]
```diff
@@ -17,6 +18,34 @@ type ShortLinkSql struct {
 	db *sql.DB
 }
 
+// UpdateOGMetaTags updates OpenGraph Meta Tags for given alias in short_link table.
+func (s *ShortLinkSql) UpdateOGMetaTags(alias string, openGraphTags metatag.OpenGraph) (entity.ShortLink, error) {
```
[the remark this answers is not in the document]
Comment (arberiii) [reply backend/app/adapter/sqldb/short_link.go:22]: @kapillamba4 In the case of changing the name of the function, please change also the title of the PR.

File: backend/app/adapter/sqldb/short_link.go [hunk 4]
```diff
@@ -17,6 +18,34 @@ type ShortLinkSql struct {
...
+		openGraphTags.Description,
+		openGraphTags.ImageURL,
+		alias,
+	)
+
+	if err != nil {
+		return entity.ShortLink{}, err
+	}
+
+	return s.GetShortLinkByAlias(alias)
```
Comment (arberiii) [inline backend/app/adapter/sqldb/short_link.go:46]: Out of curiosity, why does this function return the short link?

File: backend/app/adapter/sqldb/short_link_integration_test.go (hunk 2 above)
Comment (arberiii) [reply backend/app/adapter/sqldb/short_link_integration_test.go:90]: Please add a test case where the update tags(metaTags) provided are the same as the ones that the short link has.

Comment (byliuyang) [reply backend/app/adapter/sqldb/short_link_integration_test.go:90]: The same tags test doesn't seem to provide much value here. It's optional for @kapillamba4  to decide to add this one.

File: backend/app/adapter/sqldb/short_link.go (hunk 4 above)
Comment (kapillamba4) [reply backend/app/adapter/sqldb/short_link.go:45]: We must return something as a result of graphQL query. I thought it is better to return the ShorkLink here to verify if the update was success.
In other places we are returning the updated resource on updation:
https://github.com/short-d/short/blob/267b64bdaa039089ff49b6e63a08813daad0e917/backend/app/adapter/sqldb/short_link.go#L77

File: backend/app/adapter/sqldb/short_link.go [hunk 5]
```diff
@@ -17,6 +18,33 @@ type ShortLinkSql struct {
 	db *sql.DB
 }
 
+// UpdateOpenGraphTags updates OpenGraph Meta Tags for a given short_link.
```
Comment (byliuyang) [inline backend/app/adapter/sqldb/short_link.go:21]:
```suggestion
// UpdateOpenGraphTags updates OpenGraph meta tags for a given short link.
```

**Pull request merged**